prepare-ocp4-aws-config.yml

```diff
@@ -1,56 +1,176 @@
+      git:
+        repo: "{{ pipeline_repo_url }}"
+        dest: "{{ pipeline_repo_path }}"
+        clone: yes
+      when: is_empty_pipeline_repo_path and pipeline_repo_path == ''
+
+    # vault file needs to be included for further run of aws cluster 
+    - name: "Include vault file from pipeline"
+      include_vars:
+        file: "{{ pipeline_repo_path }}/config/ansible/vault.yml"
+
+  tasks:
     - name: "Ensure that the ocp install directory exists and is empty"
       file:
-        path: "{{ ocp_install_directory }}"
+        path: "{{ ocp_install_directory }}/{{ cluster_name }}"
         state: "{{ item }}"
       with_items:
         - absent
         - directory
 
     - name: "Inject cluster name: {{ cluster_name }} and region: {{ aws_region }} into the install config"
       template:
-        src: "{{ lookup('env', 'WORKSPACE') }}/operators/ocp-cluster-deployment/install-config.yaml.j2"
-        dest: "{{ ocp_install_directory }}/install-config.yaml"
+        src: "{{ ocp_install_config_template }}"
+        dest: "{{ ocp_install_directory }}/{{ cluster_name }}/install-config.yaml"
 
     - name: "Inject secrets into the install-config"
       blockinfile:
-        path: "{{ ocp_install_directory }}/install-config.yaml"
+        path: "{{ ocp_install_directory }}/{{ cluster_name }}/install-config.yaml"
         block: "{{ cvp_vault.ocp_40_secrets | to_nice_yaml }}"
         insertafter: EOF
         marker: ""
       no_log: true
 
-    - file:
+    - name: "Make sure the .aws directory exists insde home directory"
+      file:
         path: "{{ lookup('env', 'HOME') }}/.aws"
         state: directory
         mode: 0700
 
-    - copy:
+    - name: "Copy contents of aws_credentials into credentials folder"
+      copy:
         content: "{{ cvp_vault.aws_credentials }}"
         dest: "{{ lookup('env', 'HOME') }}/.aws/credentials"
 
-    - file:
+    - name: "Make sure the .ssh folder exists inside home directory"
+      file:
         path: "{{ lookup('env', 'HOME') }}/.ssh"
         state: directory
         mode: 0700
 
-    - copy:
+    - name: "Copy the provite key into ~/.ssh folder"
+      copy:
         content: "{{ cvp_vault.ssh_private_key }}"
         dest: "{{ lookup('env', 'HOME') }}/.ssh/private_key"
         mode: 0600
+    - block:
+      - name: "Download the ocp installer binary"
+        get_url:
+          url: "{{ ocp_installer_url }}"
+          dest: "{{ ocp_install_directory }}"
+      - name: "Unarchive the contents of openshift installer"
+        unarchive:
+          src: "{{ ocp_install_directory }}/{{ ocp_installer_url.split('/')[-1] }}"
+          dest: "{{ ocp_install_directory }}"
+      - name: "Install openshift with the generated configuration"
+        shell:
+          cmd: "openshift-install create cluster --dir {{ cluster_name }} --log-level {{ openshift_installer_log_level }}"
+        args:
+          chdir: "{{ ocp_install_directory }}"
```

Comment:
Usually it is fine to use the latest version but not in the CI :-)
There should be away how to use already existing binary e. in `/usr/local/bin/openshift-install`